packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogEventMapper.kt

```diff
@@ -0,0 +1,113 @@
+package com.datadoghq.flutter
+
+import android.os.Handler
+import android.os.Looper
+import com.datadog.android.Datadog
+import com.datadog.android.log.LogsConfiguration
+import com.datadog.android.log.model.LogEvent
+import io.flutter.plugin.common.MethodChannel
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * This is a helper class that that simplifies event mapping / scrubbing for Log events.
+ *
+ * Since it is possible for Flutter engines to shut down and be recreated, it is possible that the
+ * object used as the event mapper to be lost. This class is used as a static instance that will
+ * perform mapping and call the Flutter mapping methods using the last provided MethodChannel.
+ */
+internal class DatadogLogEventMapper {
+    private val channels: MutableSet<MethodChannel> = mutableSetOf()
+
+    fun addChannel(channel: MethodChannel) {
+        channels.add(channel)
+    }
+
+    fun removeChannel(channel: MethodChannel) {
+        channels.remove(channel)
+    }
+
+    fun attachMapper(config: LogsConfiguration.Builder): LogsConfiguration.Builder {
+        config.setEventMapper { event -> mapLogEvent(event) }
+
+        return config
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    internal fun mapLogEvent(event: LogEvent): LogEvent? {
+        val jsonEvent = event.toJson().asFlutterMap()
+        var modifiedJson: Map<String, Any?>? = null
+
+        val latch = CountDownLatch(1)
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                val channel = channels.first()
```

Comment:
it is a possible crash point here - if there is no channel by some reason, it will throw `NoSuchElementException`. So maybe you want to use `firstOrNull`? Also, to avoid concurrency, maybe it is better to get the channel reference before making `handler.post` call?

Reply:
Yeah I'll move it out for concurrency and use `firstOrNull` instead. I originally figured `first` would get caught in the try / catch there but probably better not to rely on that.